src/Microsoft.Framework.Configuration/ConfigurationSection.cs

```diff
@@ -38,64 +41,79 @@ public IEnumerable<IConfigurationSource> Sources
             }
         }
 
-        public string Get([NotNull] string key)
+        public string Key
         {
-            string value;
-            return TryGet(key, out value) ? value : null;
+            get
+            {
+                return _key;
+            }
+        }
+
+        public string Value
+        {
+            get
```

Comment:
Indexer implementation should live in common base class and do all the right argument checks.